lib/npm/publisher.js

```diff
@@ -316,47 +318,48 @@ Publisher.prototype.toNpm = function toNpm(target, payload, options) {
 
 /**
  * @function build
- *  @param {String} name - name of the package used for logging
- *  @param {Object} payload - Payload to send to carpenter to build
- *  @param {function} next - Continuation called when build finishes
+ * @param {String} name - name of the package used for logging
+ * @param {Object} payload - Payload to send to carpenter to build
+ * @param {function} next - Continuation called when build finishes
  * Trigger the Carpenter build to run, uses the npm client payload. The response is
  * not piped back to the npm client, basically making builds fire and forget.
- * @returns {Request} hyperquest request from carpenter client
  */
 Publisher.prototype.build = function build(name, payload, next) {
   var self = this;
-
-  function onError(err) {
-    self.log.error('carpenter build error for %s', name, {
-      name: name,
-      message: err.message,
-      stack: err.stack,
-      code: err.code || 'NONE'
-    });
-  }
+  var retry = new Retryme(this.retryOpts, err => err.skip);
 
   //
-  // Build verified and published package.
+  // Build verified and published package with retries.
   //
-  return self.carpenter.build({ data: { data: payload, promote: true } })
-    .once('error', onError)
-    .once('response', function response(buildLog) {
+  self.log.info('Publish package with retries.');
+
+  retry.attempt(cb => {
+    self.carpenter.build({ data: { data: payload, promote: true } })
+      .once('error', err => {
+        self.log.warn('carpenter attempts to build with retries');
```

Comment:
Can we get a more meaningful message here?

Reply:
Specifically, imagine post-mortem debugging this by looking at logs. You'd want to see something like:

```sh-session
Attempting to publish
Failed publishing due to <error>, trying again (3 retries left)
Failed publishing due to <error>, trying again (2 retries left)
Failed publishing due to <error>, trying again (1 retry left)
Failed publishing due to <error>, bailing out.
```

Reply:
@decompil3d good point! we can include how many retries left in the log, but maybe without the error cuz we will log out the error anyway if retry fails in the end.  How's your thoughts on this?